lms/djangoapps/instructor/views/instructor_dashboard.py

```diff
@@ -197,6 +197,22 @@ def instructor_dashboard_2(request, course_id):
     if certs_enabled and access['admin']:
         sections.append(_section_certificates(course))
 
+    def get_course_blocks(course_key, category):
+        """
+        Retrieve all XBlocks in the course for a particular category.
+
+        Returns only XBlocks that are published and haven't been deleted.
+        """
+        # Note: we need to check if found components have been orphaned
+        # due to a bug in split modulestore (PLAT-799).  Once that bug
+        # is resolved, we can skip the `_is_in_course_tree()` check entirely.
+        return [
+            block for block in modulestore().get_items(
```

Comment:
You can access the course block through the get_course_in_cache and then filter the result with the topological traversal, am i wrong? @andrey-canon

Reply:
this looks easier to understand

Reply:
I think it's better to use the platform API instead of accessing the modulestore directly, anyway if it works, let's keep it as it.